python-avd/pyavd/_eos_designs/shared_utils/l3_port_channels.py

```diff
@@ -0,0 +1,30 @@
+# Copyright (c) 2023-2024 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+from __future__ import annotations
+
+from functools import cached_property
+from typing import TYPE_CHECKING
+
+from pyavd._eos_designs.schema import EosDesigns
+
+if TYPE_CHECKING:
+    from . import SharedUtils
+
+
+class L3PortChannelsMixin:
+    """
+    Mixin Class providing a subset of SharedUtils.
+
+    Class should only be used as Mixin to the SharedUtils class.
+    Using type-hint on self to get proper type-hints on attributes across all Mixins.
+    """
+
+    @cached_property
+    def l3_port_channels(self: SharedUtils) -> EosDesigns._DynamicKeys.DynamicNodeTypesItem.NodeTypes.NodesItem.L3PortChannels:
+        """Returns the list of l3_port_channels."""
+        return EosDesigns._DynamicKeys.DynamicNodeTypesItem.NodeTypes.NodesItem.L3PortChannels(list(self.node_config.l3_port_channels))
```

Comment:
I don't think we need this method at all if it is not doing anything to the data. just update the calling code to read from shared_utils.node_config.l3_port_channels directly.

Reply:
Removed these properties and this newly added file as it is not needed and we can directly reference list of "l3_port_channels" as suggested in review comment.